src/cryptography/x509/base.py

```diff
@@ -723,3 +724,7 @@ def build(self, backend):
             )
 
         return backend.create_x509_revoked_certificate(self)
+
+
+def random_serial_number():
+    return utils.int_from_bytes(os.urandom(20), "big") >> 1
```

Comment:
yes, probabiliti is VERY little, but, it is reasonable to add the loop, that checks for zero and regenerate serial number. Since normally it will not be equal to zero, loop will roll only once. This small code guarantees validity (so static analyzers will be happy)